test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +92,10 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
+        if (command.isMutating()) {
```

Comment:
This test does not help much cause it could not verify only mutating commands can make change. I suggest adding an else here for other non-mutating commands to check no file was modified (e.g. check the updated time)